sdk/storage/azure-storage-queue/src/test/java/com/azure/storage/queue/spock/APISpec.groovy

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.queue.spock
+
+import com.azure.core.test.InterceptorManager
+import com.azure.core.test.TestMode
+import com.azure.core.test.utils.TestResourceNamer
+import com.azure.core.util.configuration.ConfigurationManager
+import com.azure.core.util.logging.ClientLogger
+import com.azure.storage.queue.QueueClientBuilder
+import com.azure.storage.queue.QueueServiceClient
+import com.azure.storage.queue.QueueServiceClientBuilder
+import org.junit.Rule
+import org.junit.rules.ExpectedException
+import org.junit.rules.TestName
+import spock.lang.Specification
+
+class APISpec extends Specification {
+    // Field common used for all APIs.
+    def logger = new ClientLogger(APISpec.class)
+    def AZURE_TEST_MODE = "AZURE_TEST_MODE"
+    def interceptorManager
+    def testResourceName
+
+    // Clients for API tests
+    def primaryQueueServiceClient
+    def primaryQueueServiceAsyncClient
+    def primaryQueueClient
+    def primaryQueueAsyncClient
+
+
+    // Test name for test method name.
+    @Rule
+    TestName testName = new TestName()
+    @Rule
+    ExpectedException thrown = ExpectedException.none()
+    def testMode = getTestMode()
+    def connectionString = ConfigurationManager.getConfiguration().get("AZURE_STORAGE_CONNECTION_STRING")
+
+
+    /**
+     * Setup the QueueServiceClient and QueueClient common used for the API tests.
+     */
+    def setup() {
```

Comment:
Do we need to construct both a `QueueServiceClient` and `QueueClient` in this setup?

Instead could we have the base information determined in this setup and have the subclasses have setups that construct what they need?

Same question goes for cleanup.

Reply:
Agreed for setup. We can move the client initialization to each class.
For clean up. it only does the clean up in record mode. And it is necessary to have it for every single tests. I don't think it need to refactor